pgjdbc/src/main/java/org/postgresql/PGConnection.java

```diff
@@ -32,6 +32,14 @@
   public PGNotification[] getNotifications() throws SQLException;
 
   /**
+   * Add a notification listener
+   *
+   * @param pgNotificationListener will be fired upon notification
+   * @since 9.4.1209
+   */
+  public void addNotificationListener(PGNotificationListener pgNotificationListener);
```

Comment:
Have you considered `addNotificationListener(PGNotificationListener pgNotificationListener, String... channels)` kind of API?

That would enable efficient dispach of the notifications (i.e. something like `Map<String, PgNotificationListener>`)